clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -32,8 +33,29 @@
     public long nanoseconds();
 
     /**
+     * Returns time in milliseconds as does System.currentTimeMillis(),
+     * but uses elapsed time from an arbitrary epoch more like System.nanoTime().
+     * The difference is that if somebody changes the system clock,
+     * Time.currentElapsedTime will change but nanoTime won't.
+     * @return  The time in milliseconds from some arbitrary point in time.
+     */
+    public long currentElapsedTime();
```

Comment:
These new methods are only used in the `Login` class, so I will move them there and make them private for now.